etcd/etcd.go

```diff
@@ -0,0 +1,214 @@
+// Copyright 2023 CloudWeGo Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package etcd
+
+import (
+	"bytes"
+	"context"
+	"strconv"
+	"sync"
+	"text/template"
+
+	"go.etcd.io/etcd/api/v3/mvccpb"
+
+	clientv3 "go.etcd.io/etcd/client/v3"
+
+	"github.com/cloudwego/kitex/pkg/klog"
+	"go.uber.org/zap"
+)
+
+var (
+	m      sync.Mutex
+	ctxMap = make(map[string]context.CancelFunc)
+	Num    int64
```

Comment:
Num shouldn't be exposed, otherwise it can be modified by anyone which will break the design.